src/app/shared/models/user.model.ts

```diff
@@ -8,4 +8,12 @@ export interface User {
   email?: string;
   phoneNumber?: string;
   role: string;
-}
\ No newline at end of file
+
+  constructor(info, id?: string) {
+    this.firstName = info.firstName;
+    this.lastName = info.lastName;
+    this.middleName = info.middleName;
+    this.phoneNumber = info.phoneNumber;
+    this.id = id;
```

Comment:
if id is not passed it could be set as undefined, so it needs some condition here

Reply:
then I have changed it to the required because we don't create an initial user, we always get it from the back-end with id by default. So, when we edit the user, it for sure has its id.